src/menu/charprofile.c

```diff
@@ -0,0 +1,409 @@
+		.description = "Species: Insect\nArea of Study: Evolutionary Socio-Entomology\n\nA bright bug - or was it insect? - far from her usual stomping grounds.\nShe feels that Insects have lost their “rightful place” in history.\nIf she thinks she has a raw evolutionary deal, someone ought to tell her about the trilobites…",
+		.unlock = PBGM_STAGE3_BOSS,
+		.background = "stage3/spellbg2",
+		.faces = FACES("normal", "calm", "outraged", "proud", "defeated"),
+	},
+	[PROFILE_KURUMI] = {
+		.name = "kurumi",
+		.fullname = "Kurumi",
+		.title = "High-Society Phlebotomist",
+		.description = "Species: Vampire\nArea of Study: Hematology\n\nVampiric blast from the past. Doesn’t she know Gensōkyō already has new bloodsuckers in town?\n\nSucking blood is what she’s good at, but her current job is a gatekeeper, and her true passion is high fashion.\n\nEveryone’s got multiple side-hustles these days…",
+		.unlock = PBGM_STAGE4_BOSS,
+		.background = "stage4/kurumibg1",
+		.faces = FACES("normal", "dissatisfied", "puzzled", "tsun", "tsun_blush", "defeated"),
+	},
+	[PROFILE_IKU] = {
+		.name = "iku",
+		.fullname = "Nagae Iku",
+		.title = "Fulminologist of the Heavens",
+		.description = "Species: Oarfish\nArea of Study: Meteorology (Fulminology)\n\nMessenger from the clouds, and an amateur meteorologist.\n\nSeems to have been reading a few too many outdated psychiatry textbooks in recent times.\n\nDespite being so formal and stuffy, she seems to be the only one who knows what’s going on.",
+		.background = "stage5/spell_lightning",
+		.unlock = PBGM_STAGE5_BOSS,
+		.faces = FACES("normal", "smile", "serious", "eyes_closed", "defeated"),
+	},
+	[PROFILE_ELLY] = {
+		.name = "elly",
+		.fullname = "Elly",
+		.title = "The Theoretical Reaper",
+		.description = "Species: Shinigami(?)\nArea of Study: Theoretical Physics / Forensic Pathology (dual-major)\n\nSlightly upset over being forgotten.\n\nHas apparently gotten herself a tutor in the “dark art” of theoretical physics. Her side-hustle is megalomania.\n\nLiterally on top of the world, without a care to spare for those beneath her. Try not to let too many stones crush the innocent yōkai below when her Tower of Babel starts to crumble.",
+		.background = "stage6/spellbg_toe",
+		.unlock = PBGM_STAGE6_BOSS1,
+		.faces = FACES("normal", "eyes_closed", "angry", "shouting", "smug", "blush"),
+	},
+	[PROFILE_LOCKED] = {
+		.name = "locked",
+		.fullname = "Locked",
+		.title = "...",
+		.description = "You have not unlocked this character yet!",
+		.unlock = PBGM_GAMEOVER,
+		.background = "stage1/cirnobg",
+		.faces = FACES("locked"),
+	}
+};
+
+
+static void update_char_draw_order(MenuData *m) {
+	CharProfileContext *ctx = m->context;
+
+	for(int i = 0; i < NUM_PROFILES; ++i) {
+		if(ctx->char_draw_order[i] == m->cursor) {
+			while(i) {
+				ctx->char_draw_order[i] = ctx->char_draw_order[i - 1];
+				ctx->char_draw_order[i - 1] = m->cursor;
+				--i;
+			}
+
+			break;
+		}
+	}
+}
```

Comment:
This seems to be copypasted from the charselect code but not actually used. In charselect the draw order matters because there we actually draw multiple characters at the same time over each other, since they have a slide-out animation (try to play around and switch between characters at different speeds in charselect to see how the animation behaves). Here we have a much simpler animation - only one character is displayed at a time and there's no slide-out; the previously selected character just disappears.

While it would be nice to implement a similar animation here for consistency, I'm not going to block the PR on it because it's a little complicated and can be added later. If it's not going to be implemented in this PR, then it's better to either remove the draw order stuff entirely, or keep it but add a TODO comment about this in the drawing code.